cdc/redo/writer/file.go

```diff
@@ -228,9 +231,22 @@ func (w *Writer) Write(rawData []byte) (int, error) {
 	}
 
 	n, err := w.bw.Write(rawData)
+	if err != nil {
+		return 0, err
+	}
 	w.metricWriteBytes.Add(float64(n))
-	w.size += int64(n)
-	return n, err
+
+	offset, err := w.file.Seek(0, io.SeekCurrent)
+	if err != nil {
+		return 0, err
+	}
+
+	w.curOff = offset
+	if offset < defaultMaxLogSize {
+		return n, nil
+	}
+
+	return 0, w.rotate()
```

Comment:
if rotate ,should we return 0 here ?

Reply:
Ditto, the rotate behavior is consistent with the previous implementation.